### ISSUE TITLE
asyncio.wait_for is still confusing

### ISSUE THREAD

**Issue by dce616f4-6a4f-4545-abb0-1f5d9e619b2d** [python/cpython#81917]
BPO | [37736](https://bugs.python.org/issue37736)
--- | :---
Nosy | @asvetlov, @cjrh, @1st1

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2019-08-01.09:25:01.672>
labels = ['3.8', 'type-bug', '3.7', 'expert-asyncio']
title = 'asyncio.wait_for is still confusing'
updated_at = <Date 2021-02-23.15:56:00.528>
user = 'https://bugs.python.org/DavidLewis'
```

bugs.python.org fields:
```python
activity = <Date 2021-02-23.15:56:00.528>
actor = 'andrewborba'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = ['asyncio']
creation = <Date 2019-08-01.09:25:01.672>
creator = 'David Lewis'
dependencies = []
files = []
hgrepos = []
issue_num = 37736
keywords = []
message_count = 3.0
messages = ['348848', '354029', '387578']
nosy_count = 5.0
nosy_names = ['asvetlov', 'cjrh', 'yselivanov', 'David Lewis', 'andrewborba']
pr_nums = []
priority = 'normal'
resolution = None
stage = None
status = 'open'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue37736'
versions = ['Python 3.7', 'Python 3.8']
```

</p></details>

**Comment by kumaraditya303** [comment]
Fixed by #96764

**Comment by kumaraditya303** [comment]
Catching `CancelledError` isn't recommended as it breaks `asyncio` internal timeout handling.
